js/app.js

```diff
@@ -0,0 +1,110 @@
+const drawGame = {
+    appName: 'Island Race Car',
+    author: 'Guillermo Pérez',
+    version: '1.0.0',
+    license: undefined,
+    size: { 
+        w: undefined, 
+        h: undefined 
+    },
+    canvas: document.querySelector('#canvas'),
+    ctx: undefined,
+    car: undefined,
+    arr: [],
+    framesIndex: 0,
+    init() {
+        this.setContext();
+        this.setSize();
+        this.createCar();
+        this.drawAll();
+        this.setEventHandlers();
+    },
+    setContext() {
+        this.ctx = this.canvas.getContext('2d');
+    },
+    setSize() {
+        this.size = {
+            w: window.innerWidth,
+            h: window.innerHeight
+        }
+        this.canvas.setAttribute('width', this.size.w / 2);
+        this.canvas.setAttribute('height', this.size.h);
+    },
+    drawRectangle() {
+        this.ctx.fillStyle = 'grey';
+        this.ctx.fillRect(50, 0, this.canvas.width - 100, this.canvas.height);
+    },
+    drawRegularLines() {
+        this.ctx.lineWidth = 20;
+        this.ctx.strokeStyle = 'white';
+        this.ctx.beginPath();
+        this.ctx.moveTo(80, 0);
+        this.ctx.lineTo(80, this.canvas.width);
+        this.ctx.stroke();
+        this.ctx.closePath();
+
+        this.ctx.lineWidth = 20;
+        this.ctx.strokeStyle = 'white';
+        this.ctx.beginPath();
+        this.ctx.moveTo(690, 0);
+        this.ctx.lineTo(690, this.canvas.width);
+        this.ctx.stroke();
+        this.ctx.closePath();
+    },
+    drawDashedLines() {
+        this.ctx.lineWidth = 10;
+        this.ctx.strokeStyle = 'white';
+        this.ctx.beginPath();
+        this.ctx.setLineDash([60, 20]);
+        this.ctx.moveTo(370, 0);
+        this.ctx.lineTo(370, this.canvas.width);
+        this.ctx.stroke();
+        this.ctx.closePath();
+        this.ctx.setLineDash([0, 0]);
+    },
+    pressStart() {
+        this.init();
+        this.collision();
+        this.framesIndex;
+        this.framesIndex % 50 === 0 ? this.createObstacles() : null;
+        setInterval(() => {
+            this.arr.forEach(elm => {
+                elm.move();
+                elm.draw();
+            });
+        }, 50);
+    },
+    createCar() {
+        this.car = new Car(this.ctx, 325, 550, 80);
+    },
+    createObstacles() {
+        this.arr.push(new Obstacles(this.ctx, 350, this.size));
+    },
+    collision() {
+        if (this.createCar().carPos.x < this.createObstacles().obsPos.x + this.createObstacles().obsSize.w &&
+        this.createCar().carPos.x + this.createCar().carSize.w >  this.createObstacles().obsPos.x &&
+        this.createCar().carPos.y < this.createObstacles().obsPos.y + this.createObstacles().obsSize.h &&
+        this.createCar().carSize.h + this.createCar().carPos.y > this.createObstacles().obsPos.y) {
+            alert('COLISIOOOOON');
+        }
+    },
```

Comment:
- Aquí en vez de comprobar la posición del coche de tu objeto (`this.car`), estas cada vez llamando a la instancia que lo crea, por lo tanto no puede acceder a la `carPos.x` y salta un error.
- Lo mismo pasa con los obstáculos, estás creando un obstáculo cada vez que compruebas la posición, más que comprobar la posición de los obstáculos que tienes en tu arr. 
- Una forma de solucionarlo es meter un parámetro que coja un obstáculo (el coche no haría falta porque solo tienes uno) y así comprobar las colisiones del coche con todos los obstáculos:

```suggestion
   collision(obstacle) {
        if (this.car.carPos.x < obstacle.obsPos.x + obstacle.obsSize.w &&
            this.car.carPos.x + this.car.carSize.w > obstacle.obsPos.x &&
            this.car.carPos.y < obstacle.obsPos.y + obstacle.obsSize.h &&
            this.car.carSize.h + this.car.carPos.y > obstacle.obsPos.y) {
            alert('COLISIOOOOON');
        }
    }
```